tests/epyccel/recognised_functions/test_numpy_funcs.py

```diff
@@ -3640,118 +3662,134 @@ def test_complex128(a):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_int64():
-        from numpy import real, shape, array, int64
-        arr = array([4,5,6,2,1], int64)
+    @types('int64[:]')
+    def test_int64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_float():
-        from numpy import real, shape, array
-        arr = array([4,5,6,2,1], float)
+    @types('float[:]')
+    def test_float(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_float32():
-        from numpy import real, shape, array, float32
-        arr = array([4,5,6,2,1], float32)
+    @types('float32[:]')
+    def test_float32(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_float64():
-        from numpy import real, shape, array, float64
-        arr = array([4,5,6,2,1], float64)
+    @types('float64[:]')
+    def test_float64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_complex64():
-        from numpy import real, shape, array, complex64
-        arr = array([4,5,6,2,1], complex64)
+    @types('complex64[:]')
+    def test_complex64(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
-    def test_complex128():
-        from numpy import real, shape, array, complex128
-        arr = array([4,5,6,2,1], complex128)
+    @types('complex128[:]')
+    def test_complex128(arr):
+        from numpy import real, shape
         a = real(arr)
         s = shape(a)
         return len(s), s[0], a[0]
 
+    import numpy as np
+    bl = randint(0, 1, size=(5), dtype= bool)
+
+    # should be uncommented after resolving #733
+    # integer8 = randint(np.iinfo(np.int8(1)).min, np.iinfo(np.int8(1)).max, size=(5), dtype=np.int8)
+    # integer16 = randint(np.iinfo(np.int16(1)).min, np.iinfo(np.int16(1)).max, size=(5), dtype=np.int16)
+    integer = randint(min_int, max_int, size=(5), dtype=np.int)
+    integer32 = randint(min_int32, max_int32, size=(5), dtype=np.int32)
+    integer64 = randint(min_int64, max_int64, size=(5), dtype=np.int64)
+
+    fl = uniform(min_float / 2, max_float / 2, size=(5))
+    fl32 = uniform(min_float32, max_float32, size=(5))
+    fl32 = np.float32(fl32)
+    fl64 = uniform(min_float64 / 2, max_float64 / 2, size=(5))
+
+    cmplx64 = uniform(low=min_float32 / 2, high=max_float32 / 2, size=5) + uniform(low=max_float32 / 2, high=max_float32 / 2, size=5) * 1j
+    cmplx64 = np.complex64(cmplx64)
+    cmplx128 = uniform(low=min_float64 / 2, high=max_float64 / 2, size=5) + uniform(low=min_float64 / 2, high=max_float64 / 2, size=5) * 1j
```

Comment:
try to choose meaningful names here, you may use this notation:
`arr_1d_int = randint(min_int, max_int, size=(5), dtype=np.int)`
`arr_1d_int32 = randint(min_int32, max_int32, size=(5), dtype=np.int32)`

Reply:
`arr` keyword is mentioned in the name of the test, so these names are describing just the data type that will be tested.

Reply:
take a look above you will find these variables use for single values, not arrays.

Reply:
same thing answer, those names are just to identify the type.